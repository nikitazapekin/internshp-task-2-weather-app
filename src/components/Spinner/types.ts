export interface SpinnerProps {
  position?: "relative" | "absolute";
}
