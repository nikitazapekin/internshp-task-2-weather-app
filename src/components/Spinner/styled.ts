import styled, { keyframes } from "styled-components";

const rotate = keyframes`
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
`;

export const SpinnerStyled = styled.span<{ position?: "relative" | "absolute" }>`
  width: 30px;
  height: 30px;
  border: 3px solid ${({ theme }) => theme.colors.blue};
  border-bottom-color: ${({ theme }) => theme.colors.gray};
  border-radius: 50%;
  display: inline-block;
  box-sizing: border-box;
  animation: ${rotate} 0.8s linear infinite;

  ${({ position = "absolute" }) =>
    position === "absolute"
      ? `
        position: absolute;
        left: 50%;
        top: 50%;
        transform: translateX(-50%) translateY(-50%);
      `
      : `
        position: relative;
        margin: 0 auto;
      `}
`;
