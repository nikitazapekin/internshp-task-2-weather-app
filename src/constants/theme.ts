export const theme = {
  container: 1187,
  deviceHeight: "100vh",
  colors: {
    black: "#000",
    blue: "#122e65",
    gray: "#D9D9D9",
    white: "#fff",
  },
  fontFamilies: {
    primary: "Inter",
    secondary: "sans-serif",
  },
  fontSizes: {
    xl: "41px",
    lg: "30px",
    md: "27px",
    sm: "22px",
    xs: "20px",
    xxs: "18px",
  },
  fontWeights: {
    regular: 400,
  },
  media: {
    xxl: 1400,
    xl: 1280,
    lg: 1024,
    md: 768,
    xh: 560,
    sm: 480,
  },
  zIndexes: {
    x: -1,
    xs: 0,
    sm: 1,
    md: 2,
  },
  spaces: {
    x: 5,
    xs: 10,
    sm: 15,
    md: 20,
    lg: 30,
    xl: 40,
    xxl: 50,
    g: 60,
    h: 70,
    xh: 80,
    eh: 90,
  },
  swiper: {
    item_gap: 5,
    margin_base: 2.5,
    vw_ratio: 0.5,
    scroll_snap_type: "x mandatory",
    scroll_snap_align: "start",
    flex_basis: "0 0",
    min_width: "0",
    container: 16.6,
  },
};
