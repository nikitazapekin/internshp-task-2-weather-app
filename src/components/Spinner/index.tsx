import { SpinnerStyled } from "./styled";
import type { SpinnerProps } from "./types";

const Spinner = ({ position }: SpinnerProps) => {
  return <SpinnerStyled position={position} data-testid="spinner" />;
};

export default Spinner;
