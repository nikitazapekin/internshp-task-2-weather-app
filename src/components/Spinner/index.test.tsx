import { theme } from "@constants/theme";
import { render, screen } from "@testing-library/react";
import { ThemeProvider } from "styled-components";

import Spinner from ".";

describe("Spinner Component", () => {
  test("renders spinner with correct test id", () => {
    render(
      <ThemeProvider theme={theme}>
        <Spinner position="absolute" />
      </ThemeProvider>
    );
    const spinner = screen.getByTestId("spinner");

    expect(spinner).toBeInTheDocument();
  });

  test("applies correct position style from props", () => {
    const testPosition = "relative";

    render(
      <ThemeProvider theme={theme}>
        <Spinner position={testPosition} />
      </ThemeProvider>
    );

    const spinner = screen.getByTestId("spinner");

    expect(spinner).toHaveStyle(`position: ${testPosition}`);
  });

  test("has correct default styles", () => {
    render(
      <ThemeProvider theme={theme}>
        <Spinner position="absolute" />
      </ThemeProvider>
    );

    const spinner = screen.getByTestId("spinner");

    expect(spinner).toHaveStyle("width: 30px");
    expect(spinner).toHaveStyle("height: 30px");
  });

  test("matches snapshot", () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <Spinner position="absolute" />
      </ThemeProvider>
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
