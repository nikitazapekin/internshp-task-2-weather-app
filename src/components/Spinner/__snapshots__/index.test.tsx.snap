// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Spinner Component matches snapshot 1`] = `
<span
  class="sc-bRKDuR bKSrkH"
  data-testid="spinner"
  position="absolute"
/>
`;
